depends/Makefile

```diff
@@ -73,6 +75,20 @@ include builders/$(build_os).mk
 include builders/default.mk
 include packages/packages.mk
 
+build_id_string:=$(BUILD_ID_SALT)
+build_id_string+=$(shell $(build_CC) --version 2>/dev/null)
+build_id_string+=$(shell $(build_AR) --version 2>/dev/null)
+build_id_string+=$(shell $(build_CXX) --version 2>/dev/null)
+build_id_string+=$(shell $(build_RANLIB) --version 2>/dev/null)
+build_id_string+=$(shell $(build_STRIP) --version 2>/dev/null)
```

Comment:
True - we never use `ld` directly, and we shouldn't, for example it would be quite annoying to figure out what the linker is (and what plugin it is using) in my lto setup testing the clang master branch:
```
   CC="${CLANGPATH}/bin/clang " 
   CXX="${CLANGPATH}/bin/clang++" 
   CFLAGS="-flto -O2 -g" CXXFLAGS="-flto -O2 -g" LDFLAGS="-flto -O2 -fuse-ld=gold"
   RANLIB="${CLANGPATH}/bin/llvm-ranlib"
```
So yes just wanted to ask to be sure.